packages/backend/src/modules/tournament/owner.field.ts

```diff
@@ -0,0 +1,24 @@
+import { FieldResolver, Resolver, ResolverInterface, Root } from 'type-graphql';
+import { getRepository } from 'typeorm';
+
+import DBTournament from '../../entities/tournament';
+
+import GQLTournament from './tournament';
+
+@Resolver(() => GQLTournament)
+export default class OwnerResolver implements ResolverInterface<GQLTournament> {
+	@FieldResolver()
+	async owner(@Root() { id }: GQLTournament) {
+		const tournamentRepository = getRepository(DBTournament);
+		const tournament = await tournamentRepository.findOne({
+			where: { id },
+			relations: ['owner'],
+		});
+
+		if (!tournament?.owner) {
```

Comment:
The ! is an inverse so !true === false. In this case it might seem weird but this returns true if the value is set to something else than null/undefined. So we are checking to see if there is not an owner and if tournament.owner is false(by tournament.owner being null) then we want to invert that value to true so the if-statement becomes true and we throw the error.

The questionmark is a typescript thing and means that tournament MIGHT be null. Since we don't check if we get a tournament back before we try to access it we need prepend tournament with ? to signal that it can be null and if that is the case just return null. I think if we try to access it without the ? it will error out and say "cant access owner of undefined" instead of just returning null but I am not sure. Typescript will also warn you that the value migth be null, implying you need the ?

Reply:
It's called optional chaining: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

Reply:
also, is kind of redundant here